openedx/frontend-app-course-authoring
# UI fixes for Sort Library Component [FC-0059]

## Description

Addresses issues raised during acceptance testing for https://github.com/openedx/frontend-app-course-authoring/issues/1038, specifically:

* Use "Recently Modified" as the default sort option when no keyword is entered.
* Use "Most Relevant" as the default sort option when a keyword is entered.
   Hide this sort option if no keyword is entered.
* Reorder sort options [as per pic](https://github.com/openedx/frontend-app-course-authoring/issues/1038#issuecomment-2299491683).
* Change the "Sort by" dropdown title to match the selected sort option.
* Show "Sort by" in the drop-down itself as a `Dropdown.Header`.
* Don't change the sort option when "clear filters" is clicked.
  Instead, remove a sort filter by clicking on the current "sort by" selection.
  For example, clicking on Sort > "Title, A,Z" will sort by that, clicking it again will remove that Sort and take the user back to default.

### Screenshots

See https://github.com/openedx/frontend-app-course-authoring/pull/1147 for previous screenshots.

* Default sort is "Recently Modified" (when no search keywords):

  ![default_sort](https://github.com/user-attachments/assets/cc3bac83-a247-47b2-816a-1b4ceac0bdc4)

* Default sort is "Most Relevant" (when search keywords provided):

  ![default_sort_w_keyword](https://github.com/user-attachments/assets/9a48e4e8-88d4-4606-8045-d7d389a835c5)

## Supporting information

See [discussion from comment](https://github.com/openedx/frontend-app-course-authoring/issues/1038#issuecomment-2286562482).

Private-ref: [FAL-3811](https://tasks.opencraft.com/browse/FAL-3811)

## Testing instructions

1. Create a new library
2. Add components to the library, and Publish.
4. Add more components, but leave them unpublished.

To test this change:

1. Navigate to the new library from http://apps.local.edly.io:2001/course-authoring/libraries/
2. Check that the default component sort option is "Recently Modified".
3. Enter a search keyword, and note that the default component sort option is now "Most Relevant".
4. Change the sort options and note changes in the Components list.
   For any non-default sort option, there should be a `?sort=...` entry added to the query string.
5. Click on the Components tab, and note sort option stays the same as previously selected.
6. Change sort options and note changes in the components list. 
7. Select "Recently Published" -- should omit unpublished components from the list.
8. Clicking "clear filters" should not change the selected sort option.
